### PR TITLE
fix: remove is-typedarray and typedarray-to-buffer

### PR DESCRIPTION
**Pull request by wraithgar**
our support matrix has support for sending typed arrays directly to
buffers

### CODE REVIEW

**Comment by wraithgar** [comment]
Closes https://github.com/npm/write-file-atomic/issues/86

File: lib/index.js [hunk 1]
```diff
@@ -110,10 +108,7 @@ async function writeFileAsync (filename, data, options = {}) {
     if (options.tmpfileCreated) {
       await options.tmpfileCreated(tmpfile)
     }
-    if (isTypedArray(data)) {
-      data = typedArrayToBuffer(data)
-    }
-    if (Buffer.isBuffer(data)) {
+    if (Buffer.isBuffer(data) || ArrayBuffer.isView(data)) {
```
Comment (wraithgar) [inline lib/index.js:111]: typed arrays and dataviews both return true on `isView`, fs can handle either.

Comment (jimmywarting) [reply lib/index.js:111]: yeay, this was the kind of pr i wanted to see, but just a nit pick, is it necessary to have the `Buffer.isBuffer`?
isn't using only `ArrayBuffer.isView` sufficient enough?

Comment (wraithgar) [reply lib/index.js:111]:
```js
> ArrayBuffer.isView(Buffer.from('42'))
true
```
sure enough.  Feel free to submit a pr to clean it up, I'm done for the day and won't be cutting the release for this repo till tomorrow.

**Pull request merged**